### PR TITLE
Issue #9564: update example of AST for TokenTypes.RESOURCE_SPECIFICATION

### LINKED ISSUE #9564: update example of AST for TokenTypes.RESOURCE_SPECIFICATION
https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#RESOURCE_SPECIFICATION

needs to be extended to show example of AST.
Source to be updated: https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java

PLEASE READ for more details: https://github.com/checkstyle/checkstyle/wiki/Good-practice-on-how-to-update-AST-in-TokenTypes

### PR DESCRIPTION
**Pull request by haanhvu**
closes #9564
ss1
ss2
ss3
ss4
ss5
```
> cat Test.java
import java.io.*;

public class Test {
        void method1() {
                try (Foo foo = new Foo(); Bar bar = new Bar()) {

                }
        }

        String method2() throws Exception {
                String path = "some path";
                try (BufferedReader br = new BufferedReader(new FileReader(path))) {
                        return br.readLine();
                }
        }

        private class Foo implements AutoCloseable {
                public void close() {

                }
        }

        private class Bar implements AutoCloseable{
                public void close() {

                }
        }
}

> java -jar checkstyle-8.41.1-all.jar -t Test.java
IMPORT -> import [1:0]
|--DOT -> . [1:14]
|   |--DOT -> . [1:11]
|   |   |--IDENT -> java [1:7]
|   |   `--IDENT -> io [1:12]
|   `--STAR -> * [1:15]
`--SEMI -> ; [1:16]
CLASS_DEF -> CLASS_DEF [3:0]
|--MODIFIERS -> MODIFIERS [3:0]
|   `--LITERAL_PUBLIC -> public [3:0]
|--LITERAL_CLASS -> class [3:7]
|--IDENT -> Test [3:13]
`--OBJBLOCK -> OBJBLOCK [3:18]
    |--LCURLY -> { [3:18]
    |--METHOD_DEF -> METHOD_DEF [4:1]
    |   |--MODIFIERS -> MODIFIERS [4:1]
    |   |--TYPE -> TYPE [4:1]
    |   |   `--LITERAL_VOID -> void [4:1]
    |   |--IDENT -> method1 [4:6]
    |   |--LPAREN -> ( [4:13]
    |   |--PARAMETERS -> PARAMETERS [4:14]
    |   |--RPAREN -> ) [4:14]
    |   `--SLIST -> { [4:16]
    |       |--LITERAL_TRY -> try [5:2]
    |       |   |--RESOURCE_SPECIFICATION -> RESOURCE_SPECIFICATION [5:6]
    |       |   |   |--LPAREN -> ( [5:6]
    |       |   |   |--RESOURCES -> RESOURCES [5:7]
    |       |   |   |   |--RESOURCE -> RESOURCE [5:7]
    |       |   |   |   |   |--MODIFIERS -> MODIFIERS [5:7]
    |       |   |   |   |   |--TYPE -> TYPE [5:7]
    |       |   |   |   |   |   `--IDENT -> Foo [5:7]
    |       |   |   |   |   |--IDENT -> foo [5:11]
    |       |   |   |   |   `--ASSIGN -> = [5:15]
    |       |   |   |   |       `--EXPR -> EXPR [5:17]
    |       |   |   |   |           `--LITERAL_NEW -> new [5:17]
    |       |   |   |   |               |--IDENT -> Foo [5:21]
    |       |   |   |   |               |--LPAREN -> ( [5:24]
    |       |   |   |   |               |--ELIST -> ELIST [5:25]
    |       |   |   |   |               `--RPAREN -> ) [5:25]
    |       |   |   |   |--SEMI -> ; [5:26]
    |       |   |   |   `--RESOURCE -> RESOURCE [5:28]
    |       |   |   |       |--MODIFIERS -> MODIFIERS [5:28]
    |       |   |   |       |--TYPE -> TYPE [5:28]
    |       |   |   |       |   `--IDENT -> Bar [5:28]
    |       |   |   |       |--IDENT -> bar [5:32]
    |       |   |   |       `--ASSIGN -> = [5:36]
    |       |   |   |           `--EXPR -> EXPR [5:38]
    |       |   |   |               `--LITERAL_NEW -> new [5:38]
    |       |   |   |                   |--IDENT -> Bar [5:42]
    |       |   |   |                   |--LPAREN -> ( [5:45]
    |       |   |   |                   |--ELIST -> ELIST [5:46]
    |       |   |   |                   `--RPAREN -> ) [5:46]
    |       |   |   `--RPAREN -> ) [5:47]
    |       |   `--SLIST -> { [5:49]
    |       |       `--RCURLY -> } [7:2]
    |       `--RCURLY -> } [8:1]
    |--METHOD_DEF -> METHOD_DEF [10:1]
    |   |--MODIFIERS -> MODIFIERS [10:1]
    |   |--TYPE -> TYPE [10:1]
    |   |   `--IDENT -> String [10:1]
    |   |--IDENT -> method2 [10:8]
    |   |--LPAREN -> ( [10:15]
    |   |--PARAMETERS -> PARAMETERS [10:16]
    |   |--RPAREN -> ) [10:16]
    |   |--LITERAL_THROWS -> throws [10:18]
    |   |   `--IDENT -> Exception [10:25]
    |   `--SLIST -> { [10:35]
    |       |--VARIABLE_DEF -> VARIABLE_DEF [11:2]
    |       |   |--MODIFIERS -> MODIFIERS [11:2]
    |       |   |--TYPE -> TYPE [11:2]
    |       |   |   `--IDENT -> String [11:2]
    |       |   |--IDENT -> path [11:9]
    |       |   `--ASSIGN -> = [11:14]
    |       |       `--EXPR -> EXPR [11:16]
    |       |           `--STRING_LITERAL -> "some path" [11:16]
    |       |--SEMI -> ; [11:27]
    |       |--LITERAL_TRY -> try [12:2]
    |       |   |--RESOURCE_SPECIFICATION -> RESOURCE_SPECIFICATION [12:6]
    |       |   |   |--LPAREN -> ( [12:6]
    |       |   |   |--RESOURCES -> RESOURCES [12:7]
    |       |   |   |   `--RESOURCE -> RESOURCE [12:7]
    |       |   |   |       |--MODIFIERS -> MODIFIERS [12:7]
    |       |   |   |       |--TYPE -> TYPE [12:7]
    |       |   |   |       |   `--IDENT -> BufferedReader [12:7]
    |       |   |   |       |--IDENT -> br [12:22]
    |       |   |   |       `--ASSIGN -> = [12:25]
    |       |   |   |           `--EXPR -> EXPR [12:27]
    |       |   |   |               `--LITERAL_NEW -> new [12:27]
    |       |   |   |                   |--IDENT -> BufferedReader [12:31]
    |       |   |   |                   |--LPAREN -> ( [12:45]
    |       |   |   |                   |--ELIST -> ELIST [12:46]
    |       |   |   |                   |   `--EXPR -> EXPR [12:46]
    |       |   |   |                   |       `--LITERAL_NEW -> new [12:46]
    |       |   |   |                   |           |--IDENT -> FileReader [12:50]
    |       |   |   |                   |           |--LPAREN -> ( [12:60]
    |       |   |   |                   |           |--ELIST -> ELIST [12:61]
    |       |   |   |                   |           |   `--EXPR -> EXPR [12:61]
    |       |   |   |                   |           |       `--IDENT -> path [12:61]
    |       |   |   |                   |           `--RPAREN -> ) [12:65]
    |       |   |   |                   `--RPAREN -> ) [12:66]
    |       |   |   `--RPAREN -> ) [12:67]
    |       |   `--SLIST -> { [12:69]
    |       |       |--LITERAL_RETURN -> return [13:3]
    |       |       |   |--EXPR -> EXPR [13:21]
    |       |       |   |   `--METHOD_CALL -> ( [13:21]
    |       |       |   |       |--DOT -> . [13:12]
    |       |       |   |       |   |--IDENT -> br [13:10]
    |       |       |   |       |   `--IDENT -> readLine [13:13]
    |       |       |   |       |--ELIST -> ELIST [13:22]
    |       |       |   |       `--RPAREN -> ) [13:22]
    |       |       |   `--SEMI -> ; [13:23]
    |       |       `--RCURLY -> } [14:2]
    |       `--RCURLY -> } [15:1]
    |--CLASS_DEF -> CLASS_DEF [17:1]
    |   |--MODIFIERS -> MODIFIERS [17:1]
    |   |   `--LITERAL_PRIVATE -> private [17:1]
    |   |--LITERAL_CLASS -> class [17:9]
    |   |--IDENT -> Foo [17:15]
    |   |--IMPLEMENTS_CLAUSE -> implements [17:19]
    |   |   `--IDENT -> AutoCloseable [17:30]
    |   `--OBJBLOCK -> OBJBLOCK [17:44]
    |       |--LCURLY -> { [17:44]
    |       |--METHOD_DEF -> METHOD_DEF [18:2]
    |       |   |--MODIFIERS -> MODIFIERS [18:2]
    |       |   |   `--LITERAL_PUBLIC -> public [18:2]
    |       |   |--TYPE -> TYPE [18:9]
    |       |   |   `--LITERAL_VOID -> void [18:9]
    |       |   |--IDENT -> close [18:14]
    |       |   |--LPAREN -> ( [18:19]
    |       |   |--PARAMETERS -> PARAMETERS [18:20]
    |       |   |--RPAREN -> ) [18:20]
    |       |   `--SLIST -> { [18:22]
    |       |       `--RCURLY -> } [20:2]
    |       `--RCURLY -> } [21:1]
    |--CLASS_DEF -> CLASS_DEF [23:1]
    |   |--MODIFIERS -> MODIFIERS [23:1]
    |   |   `--LITERAL_PRIVATE -> private [23:1]
    |   |--LITERAL_CLASS -> class [23:9]
    |   |--IDENT -> Bar [23:15]
    |   |--IMPLEMENTS_CLAUSE -> implements [23:19]
    |   |   `--IDENT -> AutoCloseable [23:30]
    |   `--OBJBLOCK -> OBJBLOCK [23:43]
    |       |--LCURLY -> { [23:43]
    |       |--METHOD_DEF -> METHOD_DEF [24:2]
    |       |   |--MODIFIERS -> MODIFIERS [24:2]
    |       |   |   `--LITERAL_PUBLIC -> public [24:2]
    |       |   |--TYPE -> TYPE [24:9]
    |       |   |   `--LITERAL_VOID -> void [24:9]
    |       |   |--IDENT -> close [24:14]
    |       |   |--LPAREN -> ( [24:19]
    |       |   |--PARAMETERS -> PARAMETERS [24:20]
    |       |   |--RPAREN -> ) [24:20]
    |       |   `--SLIST -> { [24:22]
    |       |       `--RCURLY -> } [26:2]
    |       `--RCURLY -> } [27:1]
    `--RCURLY -> } [28:0]
```
Expected updates for java doc are:
```
LITERAL_TRY -> try
 |--RESOURCE_SPECIFICATION -> RESOURCE_SPECIFICATION
 |   |--LPAREN -> (
 |   |--RESOURCES -> RESOURCES
 |   |   |--RESOURCE -> RESOURCE
 |   |   |   |--MODIFIERS -> MODIFIERS
 |   |   |   |--TYPE -> TYPE
 |   |   |   |   `--IDENT -> Foo
 |   |   |   |--IDENT -> foo
 |   |   |   `--ASSIGN -> =
 |   |   |       `--EXPR -> EXPR
 |   |   |           `--LITERAL_NEW -> new
 |   |   |               |--IDENT -> Foo
 |   |   |               |--LPAREN -> (
 |   |   |               |--ELIST -> ELIST
 |   |   |               `--RPAREN -> )
 |   |   |--SEMI -> ;
 |   |   `--RESOURCE -> RESOURCE
 |   |       |--MODIFIERS -> MODIFIERS
 |   |       |--TYPE -> TYPE
 |   |       |   `--IDENT -> Bar
 |   |       |--IDENT -> bar
 |   |       `--ASSIGN -> =
 |   |           `--EXPR -> EXPR
 |   |               `--LITERAL_NEW -> new
 |   |                   |--IDENT -> Bar
 |   |                   |--LPAREN -> (
 |   |                   |--ELIST -> ELIST
 |   |                   `--RPAREN -> )
 |   `--RPAREN -> )
 `--SLIST -> {
     `--RCURLY -> }
```
and
```
LITERAL_TRY -> try
 |--RESOURCE_SPECIFICATION -> RESOURCE_SPECIFICATION
 |   |--LPAREN -> (
 |   |--RESOURCES -> RESOURCES
 |   |   `--RESOURCE -> RESOURCE
 |   |       |--MODIFIERS -> MODIFIERS
 |   |       |--TYPE -> TYPE
 |   |       |   `--IDENT -> BufferedReader
 |   |       |--IDENT -> br
 |   |       `--ASSIGN -> =
 |   |           `--EXPR -> EXPR
 |   |               `--LITERAL_NEW -> new
 |   |                   |--IDENT -> BufferedReader
 |   |                   |--LPAREN -> (
 |   |                   |--ELIST -> ELIST
 |   |                   |   `--EXPR -> EXPR
 |   |                   |       `--LITERAL_NEW -> new
 |   |                   |           |--IDENT -> FileReader
 |   |                   |           |--LPAREN -> (
 |   |                   |           |--ELIST -> ELIST
 |   |                   |           |   `--EXPR -> EXPR
 |   |                   |           |       `--IDENT -> path
 |   |                   |           `--RPAREN -> )
 |   |                   `--RPAREN -> )
 |   `--RPAREN -> )
 `--SLIST -> {
     |--LITERAL_RETURN -> return
     |   |--EXPR -> EXPR
     |   |   `--METHOD_CALL -> (
     |   |       |--DOT -> .
     |   |       |   |--IDENT -> br
     |   |       |   `--IDENT -> readLine
     |   |       |--ELIST -> ELIST
     |   |       `--RPAREN -> )
     |   `--SEMI -> ;
     `--RCURLY -> }
```

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 1]
```diff
@@ -2649,117 +2649,90 @@
      *
      * <p>For example:</p>
      * <pre>
-     * try (Foo foo = new Foo(); Bar bar = new Bar()) { }
+     * try (Foo foo = new Foo(); Bar bar = new Bar()) {
+     * <br>
```
Comment (timurt) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:2653]: this tag can be removed

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 2]
```diff
@@ -2649,117 +2649,90 @@
...
+     *  `--SLIST -&gt; {
+     *      `--RCURLY -&gt; }
      * </pre>
      *
      * <p>Also consider:</p>
      * <pre>
-     * try (BufferedReader br = new BufferedReader(new FileReader(path)))
-     * {
+     * try (BufferedReader br = new BufferedReader(new FileReader(path))) {
      *  return br.readLine();
```
Comment (timurt) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:2695]: lets remove this return statement from example

**Review by timurt — CHANGES_REQUESTED** [review]
Thank you for contribution :)
please fix the following comments

**Comment by haanhvu** [comment]
@timurt I made the changes you requested, in both the code and the AST.

Please review again.

**Review by timurt — APPROVED** [review]

**Pull request merged**